main/src/main/java/com/github/topi314/lavasrc/yandexmusic/YandexMusicSourceManager.java

```
@@ -123,15 +123,15 @@ private AudioItem getAlbum(String id) throws IOException {
 
 	private AudioItem getTrack(String id) throws IOException {
 		var json = this.getJson(PUBLIC_API_BASE + "/tracks/" + id);
-		if (json.isNull() || json.get("result").values().get(0).get("available").text().equals("false")) {
+		if (json == null || json.isNull() || !json.get("error").isNull() || json.get("result").values().get(0).get("available").text().equals("false")) {
```

instead of returning `AudioReference.NO_TRACK` on error we should throw an exception